src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -369,8 +370,21 @@ private unsafe static bool IsValidHeaderString(string value)
                 return false;
             }
         }
+        private static ReadOnlySpan<byte> HexEncodeMap => new byte[] { (byte)'0', (byte)'1', (byte)'2', (byte)'3', (byte)'4', (byte)'5', (byte)'6', (byte)'7', (byte)'8', (byte)'9', (byte)'A', (byte)'B', (byte)'C', (byte)'D', (byte)'E', (byte)'F' };
 
-        private static readonly char[] s_encode16Chars = "0123456789ABCDEF".ToCharArray();
+        private static readonly SpanAction<char, (string str, char separator, uint number)> s_populateSpanWithHexSuffix = PopulateSpanWithHexSuffix;
+
+        private static readonly Vector128<byte> s_shuffleMask = Vector128.Create((byte)
```

Comment:
The field name is quite generic.
```suggestion
        private static readonly Vector128<byte> s_numberToHexShuffleMask = Vector128.Create((byte)
```
Same below for `s_asciiUpperCase` -> `s_hexAsciiUpperCase` or that like.

Personally I wouldn't use fields here and encapsulate these "constants" into the method like
```c#
private static void PopulateSpanWithHexSuffix(...)
{
    // ...

    if (Ssse3.IsSupported)
    {
        ReadOnlySpan<byte> shuffleMaskData = new byte[16]
        {
            0xF, 0xF, 3, 0xF,
            0xF, 0xF, 2, 0xF,
            0xF, 0xF, 1, 0xF,
            0xF, 0xF, 0, 0xF
        };

        ReadOnlySpan<byte> asciiUpperCaseData = new byte[16]
        {
            (byte)'0', (byte)'1', (byte)'2', (byte)'3',
            (byte)'4', (byte)'5', (byte)'6', (byte)'7',
            (byte)'8', (byte)'9', (byte)'A', (byte)'B',
            (byte)'C', (byte)'D', (byte)'E', (byte)'F'
        };

        Vector128<byte> shuffleMask = Unsafe.ReadUnaligned<Vector128<byte>>(ref MemoryMarshal.GetReference(shuffleMaskData));
        Vector128<byte> asciiUpperCase = Unsafe.ReadUnaligned<Vector128<byte>>(ref MemoryMarshal.GetReference(asciiUpperCaseData));

        var lowNibbles = // ....
    // ...
}
```

* the scope of these vector-constants is limited
* the ROS-trick will work
* less work for the JIT with static initialization --> here the vectors are just read from assembly-memory

To a good point it is matter of taste, though...

Reply:
k, this is quite nice :)